### PR TITLE
[Anomaly Task] Fix non deterministic + sample.py

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Changes

- Fix doc string in sample.py
- Add seed to train task. The only issue here is that I am not sure how the tests are run. They might not use seed. This is also the second issue. It changes the function signature which might not be a good idea.
- I've added a small readme to tools folder regarding sample.py. I feel these two commands are enough but let me know if you need anything else.

# Tests

🏃‍♂️ https://ci-ote.iotg.sclab.intel.com/job/ote/job/ote-test/1820/

### CODE REVIEW

File: external/anomaly/tools/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+python create_mvtec_ad_json_annotations.py --data_path ~/datasets/MVTec --annotation_path ~/training_extensions/data/MVtec/
+```
+
+This will generate three folders in `~/training_extensions/data/MVtec/` for classification, segmentation and detection task.
+
+Then, to run sample.py you can use the following command.
+
+```bash
+python tools/sample.py --dataset_path ~/datasets/MVTec --category bottle --train-ann-files ../../data/MVtec/bottle/segmentation/train.json --val-ann-files ../../data/MVtec/bottle/segmentation/val.json --test-ann-files ../../data/MVtec/bottle/segmentation/test.json --model_template_path ./configs/anomaly_segmentation/padim/template.yaml
+```
```
Comment (samet-akcay) [inline external/anomaly/tools/README.md]: Can you split the line here

```
python tools/sample.py \
    --dataset_path ~/datasets/MVTec \
    --category bottle \
    --train-ann-files ../../data/MVtec/bottle/segmentation/train.json \
    --val-ann-files ../../data/MVtec/bottle/segmentation/val.json \
    --test-ann-files ../../data/MVtec/bottle/segmentation/test.json \
    --model_template_path ./configs/anomaly_segmentation/padim/template.yaml
```

File: external/anomaly/tools/sample.py [hunk 2]
```diff
@@ -61,13 +61,18 @@ def __init__(
         val_subset: Dict[str, str],
         test_subset: Dict[str, str],
         model_template_path: str,
+        seed: Optional[int] = 0,
```
Comment (samet-akcay) [inline external/anomaly/tools/sample.py]: If the variable `Optional`, shouldn't it be set to `None`?

File: external/anomaly/ote_anomalib/train_task.py [hunk 3]
```diff
@@ -36,17 +38,26 @@ def train(
         dataset: DatasetEntity,
         output_model: ModelEntity,
         train_parameters: TrainParameters,
+        seed: Optional[int] = 0,
```
Comment (samet-akcay) [inline external/anomaly/ote_anomalib/train_task.py]: If the variable `Optional`, shouldn't it be set to `None`?

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Thanks for creating this PR, only minor comments

**Comment by ashwinvaidya17** [comment]
Anomaly tests have passed but looks like deep-object-reid and mmsegmentation tests are failing

**Review by goodsong81 — APPROVED** [review]
LGTM

**Pull request merged**